scripts/build-types.ts

```diff
@@ -214,13 +222,34 @@ export type RequestMessage<T = keyof OBSRequestTypes> = T extends keyof OBSReque
 	requestData: OBSRequestTypes[T];
 } : never;
 
+export type RequestBatchRequest<T = keyof OBSRequestTypes> = T extends keyof OBSRequestTypes ? OBSRequestTypes[T] extends never ? {
+	requestType: T;
+	requestId?: string;
+} : {
+	requestType: T;
+	requestId?: string;
+	requestData: OBSRequestTypes[T];
+} : never;
+
+export type RequestBatchMessage = {
+	requestId: string;
+	haltOnFailure?: boolean;
+	executionType?: RequestBatchExecutionType
+	requests: RequestBatchRequest[];
+};
```

Comment:
Maybe extract the extra options into it's own type so it could also be referred to in `callBatch` method

```suggestion
export interface RequestBatchOptions {
	/**
	* The mode of execution obs-websocket will run the batch in
	*/
	executionType?: RequestBatchExecutionType
	/**
	* Whether obs-websocket should stop executing the batch if one request fails
	*/
	haltOnFailure?: boolean;
}

export type RequestBatchMessage = {
	requestId: string;
	requests: RequestBatchRequest[];
};
```